kubeflow/kubebench/prototypes/kubebench-job.jsonnet

```diff
@@ -0,0 +1,87 @@
+// @optionalParam controllerImage string gcr.io/xyhuang-kubeflow/kubebench-controller:v20180913-1 Configurator image
+// @optionalParam githubTokenSecret string null Github token secret
+// @optionalParam githubTokenSecretKey string null Key of Github token secret
+// @optionalParam gcpCredentialsSecret string null GCP credentials secret
+// @optionalParam gcpCredentialsSecretKey string null Key of GCP credentials secret
+// @optionalParam mainJobKsPrototype string kubebench-example-tfcnn The Ksonnet prototype of the job being benchmarked
+// @optionalParam mainJobKsPackage string kubebench-examples The Ksonnet package of the job being benchmarked
+// @optionalParam mainJobKsRegistry string github.com/kubeflow/kubebench/tree/master/kubebench The Ksonnet registry of the job being benchmarked
+// @optionalParam mainJobConfig string tf-cnn-dummy.yaml Path to the config of the benchmarked job
+// @optionalParam experimentConfigPvc string kubebench-config-pvc Configuration PVC
+// @optionalParam experimentDataPvc string null Data PVC
+// @optionalParam experimentRecordPvc string kubebench-exp-pvc Experiment PVC
+// @optionalParam postJobImage string gcr.io/xyhuang-kubeflow/kubebench-example-tfcnn-postprocessor:v20180909-1 Image of post processor
+// @optionalParam postJobArgs string null Arguments of post processor
+// @optionalParam reporterType string csv Type of reporter
+// @optionalParam csvReporterInput string result.json The input of CSV reporter
+// @optionalParam csvReporterOutput string report.csv The output of CSV reporter
+
+local k = import "k.libsonnet";
+local kubebenchJob = import "kubeflow/kubebench/kubebench-job.libsonnet";
+local kubebenchRbac = import "kubeflow/kubebench/kubebench-rbac.libsonnet";
+
+local name = params.name;
+local namespace = if params.namespace == "null" then env.namespace else params.namespace;
+local serviceAccount = params.serviceAccount;
+local controllerImage = params.controllerImage;
+local configPvc = params.experimentConfigPvc;
+local dataPvc = params.experimentDataPvc;
+local experimentPvc = params.experimentRecordPvc;
+local gcpCredentialsSecret = params.gcpCredentialsSecret;
+local gcpCredentialsSecretKey = params.gcpCredentialsSecretKey;
+local githubTokenSecret = params.githubTokenSecret;
+local githubTokenSecretKey = params.githubTokenSecretKey;
+local mainJobKsPrototype = params.mainJobKsPrototype;
+local mainJobKsPackage = params.mainJobKsPackage;
+local mainJobKsRegistry = params.mainJobKsRegistry;
+local mainJobConfig = params.mainJobConfig;
+local postJobArgsStr = params.postJobArgs;
+local postJobImage = params.postJobImage;
+local reporterType = params.reporterType;
+local csvReporterInput = params.csvReporterInput;
+local csvReporterOutput = params.csvReporterOutput;
+
+local postJobArgs =
+  if postJobArgsStr == "null" then
+    []
+  else
+    std.split(postJobArgs, ",");
+
+local newUser = if serviceAccount == "null" then "kubebench-user-" + name else serviceAccount;
```

Comment:
this variable newUser is actually the service account name to use right?
Should we rename it?

Reply:
yes i will rename this, thanks.

Reply:
thanks